Patch migrated from Desktop, where it was listed as:
# owner:laca type:bug date:2005-10-13

After talking with Rod, it seems like this should apply to all platforms;
not just Solaris, otherwise there could be a significant performance
penalty at startup.  Need to file upstream bug for that.

--- a/gmodule/gmodule-dl.c	2019-09-25 15:16:54.477612239 -0700
+++ b/gmodule/gmodule-dl.c	2019-09-25 15:18:01.146526372 -0700
@@ -133,8 +133,17 @@
   gpointer handle;
   
   lock_dlerror ();
+
+#if defined(sun) && defined(__SVR4)
+  /*
+   * Always use RTLD_LAZY on Solaris; otherwise all relocations are performed
+   * immediately in all dynamic dependencies.
+   */
+  bind_lazy = 1;
+#endif
+
   handle = dlopen (file_name,
-		   (bind_local ? 0 : RTLD_GLOBAL) | (bind_lazy ? RTLD_LAZY : RTLD_NOW));
+		   (bind_local ? 0 : RTLD_GLOBAL) | (bind_lazy ? RTLD_LAZY | RTLD_FIRST : RTLD_NOW));
   if (!handle)
     {
       const gchar *message = fetch_dlerror (TRUE);
